x-pack/legacy/plugins/ml/public/timeseriesexplorer/timeseriesexplorer_utils/get_focus_data.ts

```diff
@@ -0,0 +1,169 @@
+import { mlResultsService, CriteriaField } from '../../services/results_service';
+import { Job } from '../../jobs/new_job/common/job_creator/configs';
+import { MAX_SCHEDULED_EVENTS, TIME_FIELD_NAME } from '../timeseriesexplorer_constants';
+import {
+  processDataForFocusAnomalies,
+  processForecastResults,
+  processMetricPlotResults,
+  processScheduledEventsForChart,
+} from './timeseriesexplorer_utils';
+import { mlForecastService } from '../../services/forecast_service';
+import { mlFunctionToESAggregation } from '../../../common/util/job_utils';
+import { Annotation } from '../../../common/types/annotations';
+
+const mlAnnotationsEnabled = chrome.getInjected('mlAnnotationsEnabled', false);
+
+export interface Interval {
+  asMilliseconds: () => number;
+  expression: string;
+}
+
+export interface FocusData {
+  focusChartData: any;
+  anomalyRecords: any;
+  scheduledEvents: any;
+  showForecastCheckbox: any;
+  focusAnnotationData: any;
+  focusForecastData: any;
+}
+
+export function getFocusData(
+  criteriaFields: CriteriaField[],
+  detectorIndex: number,
+  focusAggregationInterval: Interval,
+  forecastId: string,
+  modelPlotEnabled: boolean,
+  nonBlankEntities: any[],
+  searchBounds: any,
+  selectedJob: Job
+): Observable<FocusData> {
+  return forkJoin([
+    // Query 1 - load metric data across selected time range.
+    mlTimeSeriesSearchService.getMetricData(
+      selectedJob,
+      detectorIndex,
+      nonBlankEntities,
+      searchBounds.min.valueOf(),
+      searchBounds.max.valueOf(),
+      focusAggregationInterval.expression
+    ),
+    // Query 2 - load all the records across selected time range for the chart anomaly markers.
+    mlResultsService.getRecordsForCriteria(
+      [selectedJob.job_id],
+      criteriaFields,
+      0,
+      searchBounds.min.valueOf(),
+      searchBounds.max.valueOf(),
+      ANOMALIES_TABLE_DEFAULT_QUERY_SIZE
+    ),
+    // Query 3 - load any scheduled events for the selected job.
+    mlResultsService.getScheduledEventsByBucket(
+      [selectedJob.job_id],
+      searchBounds.min.valueOf(),
+      searchBounds.max.valueOf(),
+      focusAggregationInterval.expression,
+      1,
+      MAX_SCHEDULED_EVENTS
+    ),
+    // Query 4 - load any annotations for the selected job.
+    mlAnnotationsEnabled
+      ? ml.annotations
+          .getAnnotations({
+            jobIds: [selectedJob.job_id],
+            earliestMs: searchBounds.min.valueOf(),
+            latestMs: searchBounds.max.valueOf(),
+            maxAnnotations: ANNOTATIONS_TABLE_DEFAULT_QUERY_SIZE,
+          })
+          .pipe(
+            catchError(() => {
+              // silent fail
+              return of({ annotations: {} as Record<string, Annotation[]> });
+            })
+          )
+      : of(null),
+    // Plus query for forecast data if there is a forecastId stored in the appState.
+    forecastId !== undefined
+      ? (function() {
```

Comment:
nit, could be an arrow function

Reply:
https://github.com/elastic/kibana/pull/51008/commits/ac068dc12d7f26a8abf85aafd7cc5f3229862f73